plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/external/ExternalLibrariesReloadHelper.java

```diff
@@ -69,17 +69,24 @@
 	public void reloadLibraries(final boolean refreshNpmDefinitions, final IProgressMonitor monitor)
```

Comment:
According to this logics, we have to do this for ALL places in which we trigger the build.
The general question is: can multiple threads execute the build code concurrently? Probably yes at the moment.

Isn't it better to make the build code its self thread-safe with e.g. `synchronized`?

Reply:
Consider using a workspace root scheduling in `ExternalLibraryBuilderHelper`.

Reply:
re 1st comment: as discussed (would not be in line with how Eclipse invokes the builder usually)

re 2nd comment: good idea! Will check that.